cmd/admission-webhook/main.go

```diff
@@ -51,10 +57,32 @@ func main() {
 	}
 	version.LogVersionInfo()
 
-	podAh := &webhook.PodAdmissionHook{
+	ah := &webhook.AdmissionHook{
 		ExtraServiceAccounts:     extraServiceAccounts,
 		EvictRegionLeaderTimeout: evictRegionLeaderTimeout,
 	}
-	stsAh := &webhook.StatefulSetAdmissionHook{}
-	cmd.RunAdmissionServer(podAh, stsAh)
+	run(flagset, ah)
+}
+
+func run(flagset *flag.FlagSet, admissionHooks ...apiserver.AdmissionHook) {
+
+	if len(os.Getenv("GOMAXPROCS")) == 0 {
+		runtime.GOMAXPROCS(runtime.NumCPU())
+	}
+
+	stopCh := genericapiserver.SetupSignalHandler()
+
+	cmd := server.NewCommandStartAdmissionServer(os.Stdout, os.Stderr, stopCh, admissionHooks...)
+
+	// Add admission hook flags
+	cmd.Flags().AddGoFlagSet(flagset)
+
+	// Flags for glog
+	cmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
+	// Fix glog printing "Error: logging before flag.Parse"
```

Comment:
Does the klog still have this issue?

Reply:
updated, this logic was useless after testing and removed it now.